### PR TITLE
PLAT-494 Describe architecture in README.md

### PR DESCRIPTION
**Pull request by alexander-schmidt**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -144,7 +144,24 @@ _SoftiCAR GitHub Runner_ therefore employs [Sonatype Nexus](https://github.com/s
...
 
-TODO
+The basic _SoftiCAR GitHub Runner_ architecture is depicted below:
+
+![](./docs/softicar-github-runner-architecture.png)
+
+Notes on components:
+
+- `runner` container:
+  - Uses the `sysbox` Docker runtime
```
Comment (richers) [inline README.md:154]:
```suggestion
  - Uses the `sysbox` Docker runtime to enable DinD
```

File: README.md [hunk 2]
```diff
@@ -144,7 +144,24 @@ _SoftiCAR GitHub Runner_ therefore employs [Sonatype Nexus](https://github.com/s
...
+  - Is unprivileged
+  - Has restricted network access
+  - Has no bind-mounts with write access
+  - Is disposed and recreated after each build
+- `nexus` container:
+  - Serves as pull-through cache proxy for build-time resources (i.e. Docker images, Gradle plugins and Gradle dependencies)
+  - Avoids repeated downloads after the `runner` container was disposed
+- `Docker Engine (nested)`:
+  - Runs in an unprivileged container
+  - This is enabled via the `sysbox` runtime of `runner` container
```
Comment (richers) [inline README.md:164]:
```suggestion
  - This is enabled via the `sysbox` runtime of the `runner` container
```

File: README.md [hunk 3]
```diff
@@ -144,7 +144,24 @@ _SoftiCAR GitHub Runner_ therefore employs [Sonatype Nexus](https://github.com/s
...
+![](./docs/softicar-github-runner-architecture.png)
+
+Notes on components:
+
+- `runner` container:
+  - Uses the `sysbox` Docker runtime
+  - Is unprivileged
+  - Has restricted network access
+  - Has no bind-mounts with write access
+  - Is disposed and recreated after each build
```
Comment (richers) [inline README.md:158]:
```suggestion
  - Is ephemeral, i.e. is disposed and recreated after each build
```

**Review by richers — CHANGES_REQUESTED** [review]

File: README.md (hunk 3 above)
Comment (alexander-schmidt) [reply README.md:158]: This is problematic: In the context of GitHub Actions Runner, "ephemeral" only means that the runner de-registers itself after a single action (build). It does _not_ imply disposal and/or recreation. That's why the term was avoided here.

**Review by richers — APPROVED** [review]

**Pull request merged**